canon/src/implementations.rs

```diff
@@ -150,6 +150,20 @@ impl<S: Store> Canon<S> for () {
     }
 }
 
+impl<S: Store> Canon<S> for ! {
+    fn write(&self, _: &mut impl Sink<S>) -> Result<(), S::Error> {
+        loop {}
```

Comment:
Could be nice to add an explanation on why `loop{}`

Reply:
done